src/Client.spec.ts

```diff
@@ -0,0 +1,44 @@
+
+import {Client, IRequest, IRequestHeaders} from "./Client";
+describe("Client", () => {
+    it("Should be defined", () => {
+        const client = new Client();
+        expect(client).toBeDefined();
+    });
+    it("Should accept IRequest as default option", () => {
+        const headers: IRequestHeaders = {
+            "Accept": "",
+            "Authorization": "",
+            "Content-Type": "application/json",
+            "Accept-Language": "en"
+        };
+        const request: IRequest = {
+            baseUrl: "http://api.example.com/v1?user=1",
+            method: "GET",
+            queryParameters: {user: "1", id: "2"},
+            headers
+        };
+        const client = new Client(request);
+        expect(client).toBeDefined();
```

Comment:
This test should check the returned defaultOptions of client to be sure that the pass-through actually works.